ta/pkcs11/src/processing_symm.c

```diff
@@ -0,0 +1,429 @@
+		{ PKCS11_CKK_MD5_HMAC, TEE_TYPE_HMAC_MD5 },
+		{ PKCS11_CKK_SHA_1_HMAC, TEE_TYPE_HMAC_SHA1 },
+		{ PKCS11_CKK_SHA224_HMAC, TEE_TYPE_HMAC_SHA224 },
+		{ PKCS11_CKK_SHA256_HMAC, TEE_TYPE_HMAC_SHA256 },
+		{ PKCS11_CKK_SHA384_HMAC, TEE_TYPE_HMAC_SHA384 },
+		{ PKCS11_CKK_SHA512_HMAC, TEE_TYPE_HMAC_SHA512 },
+	};
+	size_t n = 0;
+	enum pkcs11_key_type key_type = get_key_type(obj->attributes);
+
+	assert(get_class(obj->attributes) == PKCS11_CKO_SECRET_KEY);
+
+	for (n = 0; n < ARRAY_SIZE(pkcs2tee_key_type); n++) {
+		if (pkcs2tee_key_type[n].key_type == key_type) {
+			*tee_type = pkcs2tee_key_type[n].tee_id;
+			return PKCS11_CKR_OK;
+		}
+	}
+
+	return PKCS11_RV_NOT_FOUND;
+}
+
+static enum pkcs11_rc
+allocate_tee_operation(struct pkcs11_session *session,
+		       enum processing_func function,
+		       struct pkcs11_attribute_head *params,
+		       struct pkcs11_object *obj)
+{
+	uint32_t size = (uint32_t)get_object_key_bit_size(obj);
+	uint32_t algo = 0;
+	uint32_t mode = 0;
+	TEE_Result res = TEE_ERROR_GENERIC;
+
+	assert(session->processing->tee_op_handle == TEE_HANDLE_NULL);
+
+	if (pkcs2tee_algorithm(&algo, params))
+		return PKCS11_CKR_FUNCTION_FAILED;
+
+	/* Sign/Verify with AES or generic key relate to TEE MAC operation */
+	pkcs2tee_mode(&mode, function);
+
+	res = TEE_AllocateOperation(&session->processing->tee_op_handle,
+				    algo, mode, size);
+	if (res)
+		EMSG("TEE_AllocateOp. failed %#"PRIx32" %#"PRIx32" %#"PRIx32,
+		     algo, mode, size);
+
+	if (res == TEE_ERROR_NOT_SUPPORTED)
+		return PKCS11_CKR_MECHANISM_INVALID;
+
+	return tee2pkcs_error(res);
+}
+
+static enum pkcs11_rc load_tee_key(struct pkcs11_session *session,
+				   struct pkcs11_object *obj)
+{
+	TEE_Attribute tee_attr;
+	size_t object_size = 0;
+	uint32_t tee_key_type = 0;
+	enum pkcs11_rc rc = PKCS11_CKR_OK;
+	TEE_Result res = TEE_ERROR_GENERIC;
+
+	TEE_MemFill(&tee_attr, 0, sizeof(tee_attr));
+
+	if (obj->key_handle != TEE_HANDLE_NULL) {
+		/* Key was already loaded and fits current need */
+		goto key_ready;
+	}
+
+	if (!pkcs2tee_load_attr(&tee_attr, TEE_ATTR_SECRET_VALUE,
+				obj, PKCS11_CKA_VALUE)) {
+		EMSG("No secret found");
+		return PKCS11_CKR_FUNCTION_FAILED;
+	}
+
+	rc = pkcs2tee_key_type(&tee_key_type, obj);
+	if (rc)
+		return rc;
+
+	object_size = get_object_key_bit_size(obj);
+	if (!object_size)
+		return PKCS11_CKR_GENERAL_ERROR;
+
+	res = TEE_AllocateTransientObject(tee_key_type, object_size,
+					  &obj->key_handle);
+	if (res) {
+		DMSG("TEE_AllocateTransientObject failed, %#"PRIx32, res);
+		return tee2pkcs_error(res);
+	}
+
+	res = TEE_PopulateTransientObject(obj->key_handle, &tee_attr, 1);
+	if (res) {
+		DMSG("TEE_PopulateTransientObject failed, %#"PRIx32, res);
+		goto error;
+	}
+
+key_ready:
+	res = TEE_SetOperationKey(session->processing->tee_op_handle,
+				  obj->key_handle);
+	if (res) {
+		DMSG("TEE_SetOperationKey failed, %#"PRIx32, res);
+		goto error;
+	}
+
+	return tee2pkcs_error(res);
```

Comment:
minor: I think returning `PKCS11_CKR_OK` straight would be better for maintenance.